plugins/telemetry-otel/src/internalClusterTest/java/org/opensearch/telemetry/tracing/TelemetryTracerEnabledSanityIT.java

```diff
@@ -0,0 +1,95 @@
+ */
+
+package org.opensearch.telemetry.tracing;
+
+import org.opensearch.common.settings.Settings;
+import org.opensearch.telemetry.TelemetrySettings;
+import org.opensearch.test.OpenSearchIntegTestCase;
+import org.opensearch.telemetry.OTelTelemetrySettings;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.client.Client;
+import org.opensearch.test.telemetry.tracing.TelemetryValidators;
+import org.opensearch.test.telemetry.tracing.validators.AllSpansAreEndedProperly;
+import org.opensearch.test.telemetry.tracing.validators.AllSpansAreInOrder;
+import org.opensearch.test.telemetry.tracing.validators.AllSpansHaveUniqueId;
+import org.opensearch.test.telemetry.tracing.validators.NumberOfTraceIDsEqualToRequests;
+import org.opensearch.test.telemetry.tracing.validators.TotalParentSpansEqualToRequests;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.opensearch.index.query.QueryBuilders.queryStringQuery;
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, minNumDataNodes = 2)
+public class TelemetryTracerEnabledSanityIT extends OpenSearchIntegTestCase {
+
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        return Settings.builder()
+            .put(super.nodeSettings(nodeOrdinal))
+            .put(
+                OTelTelemetrySettings.OTEL_TRACER_SPAN_EXPORTER_CLASS_SETTING.getKey(),
+                "org.opensearch.telemetry.tracing.InMemorySingletonSpanExporter"
+            )
+            .build();
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Arrays.asList(IntegrationTestOTelTelemetryPlugin.class);
+    }
+
+    @Override
+    protected boolean addMockTelemetryPlugin() {
+        return false;
+    }
+
+    public void testSanityChecksWhenTracingEnabled() throws Exception {
+        Client client = client();
+        // ENABLE TRACING
+        client.admin()
+            .cluster()
+            .prepareUpdateSettings()
+            .setTransientSettings(Settings.builder().put(TelemetrySettings.TRACER_ENABLED_SETTING.getKey(), true))
+            .get();
+
+        // Create Index and ingest data
+        String indexName = "test-index-11";
+        Settings basicSettings = Settings.builder().put("number_of_shards", 3).put("number_of_replicas", 0).build();
+        createIndex(indexName, basicSettings);
+        indexRandom(true, client.prepareIndex(indexName).setId("1").setSource("field1", "the fox jumps in the well"));
+        indexRandom(true, client.prepareIndex(indexName).setId("1").setSource("field2", "another fox did the same."));
+
+        ensureGreen();
+        refresh();
+
+        // Make the search calls;
+        client.prepareSearch().setQuery(queryStringQuery("fox")).get();
+        client.prepareSearch().setQuery(queryStringQuery("jumps")).get();
+
+        // Sleep for about 3s to wait for traces are published
+        Thread.sleep(3000);
```

Comment:
Can you please explicitly configure the "telemetry.otel.tracer.exporter.delay" setting to 2 seconds or less. Otherwise this might fail if default value gets changed.